File: psra/repository/src/main/java/by/arhor/psra/repository/model/Tag.java
package by.arhor.psra.repository.model;

import java.io.Serializable;

import by.arhor.psra.CoreVersion;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Tag implements Serializable, Cloneable {
	
	private static final long serialVersionUID = CoreVersion.SERIAL_VERSION_UID;
	
    private String name;
    
    @Override
    public Tag clone() {
    	return new Tag(this.name);
    }
    
}

File: psra/model/src/main/scala/by/arhor/psra/model/Tag.scala
package by.arhor.psra.model

import by.arhor.psra.CoreVersion

import scala.beans.BeanProperty

object Tag {
  val serialVersionUID: Long = CoreVersion.SERIAL_VERSION_UID
}

case class Tag(@BeanProperty var name: String) extends Serializable {

  override def toString: String = s"${getClass.getSimpleName} [name=$name]"

}
